packages/cli/src/commands/bisect/index.ts

```diff
@@ -0,0 +1,411 @@
+  let subpath = argv['--path'] || '';
+  let run = argv['--run'] || '';
+  const openEnabled = argv['--open'] || false;
+
+  if (run) {
+    run = resolve(run);
+  }
+
+  if (!bad.startsWith('https://')) {
+    bad = `https://${bad}`;
+  }
+  let parsed = parse(bad);
+  if (!parsed.hostname) {
+    output.error('Invalid input: no hostname provided');
+    return 1;
+  }
+  bad = parsed.hostname;
+  if (typeof parsed.path === 'string' && parsed.path !== '/') {
+    if (subpath && subpath !== parsed.path) {
+      output.note(
+        `Ignoring subpath ${chalk.bold(
+          parsed.path
+        )} in favor of \`--path\` argument ${chalk.bold(subpath)}`
+      );
+    } else {
+      subpath = parsed.path;
+    }
+  }
+
+  const badDeploymentPromise = getDeployment(client, bad).catch(err => err);
+
+  if (!good.startsWith('https://')) {
+    good = `https://${good}`;
+  }
+  parsed = parse(good);
+  if (!parsed.hostname) {
+    output.error('Invalid input: no hostname provided');
+    return 1;
+  }
+  good = parsed.hostname;
+  if (
+    typeof parsed.path === 'string' &&
+    parsed.path !== '/' &&
+    subpath &&
+    subpath !== parsed.path
+  ) {
+    output.note(
+      `Ignoring subpath ${chalk.bold(
+        parsed.path
+      )} which does not match ${chalk.bold(subpath)}`
+    );
+  }
+
+  const goodDeploymentPromise = getDeployment(client, good).catch(err => err);
+
+  if (!subpath) {
+    subpath = await prompt(
+      output,
+      `Specify the URL subpath where the bug occurs:`
+    );
+  }
+
+  output.spinner('Retrieving deployments…');
+  const [badDeployment, goodDeployment] = await Promise.all([
+    badDeploymentPromise,
+    goodDeploymentPromise,
+  ]);
+
+  if (badDeployment) {
+    if (badDeployment instanceof Error) {
+      badDeployment.message += ` "${bad}"`;
+      output.prettyError(badDeployment);
+      return 1;
+    }
+    bad = badDeployment.url;
+  } else {
+    output.error(`Failed to retrieve ${chalk.bold('bad')} Deployment: ${bad}`);
+    return 1;
+  }
+
+  const { projectId } = badDeployment;
+
+  if (goodDeployment) {
+    if (goodDeployment instanceof Error) {
+      goodDeployment.message += ` "${good}"`;
+      output.prettyError(goodDeployment);
+      return 1;
+    }
+    good = goodDeployment.url;
+  } else {
+    output.error(
+      `Failed to retrieve ${chalk.bold('good')} Deployment: ${good}`
+    );
+    return 1;
+  }
+
+  if (projectId !== goodDeployment.projectId) {
+    output.error(`Good and Bad deployments must be from the same Project`);
+    return 1;
+  }
+
+  if (badDeployment.createdAt < goodDeployment.createdAt) {
+    output.error(`Good deployment must be older than the Bad deployment`);
+    return 1;
+  }
+
+  if (badDeployment.target !== goodDeployment.target) {
+    output.error(
+      `Bad deployment target "${badDeployment.target}" does not match good deployment target "${goodDeployment.target}"`
+    );
```

Comment:
We might want to use `target || 'preview'` since `target` is undefined for preview deployments